app/models/concerns/publishable.rb

```diff
@@ -20,6 +20,10 @@ module Publishable
       where condition
     }
 
+    before_validation -> {
+      self.published_on ||= Time.zone.now if published?
```

Comment:
Actually we rather try to use this technique in our project - i.e. saving all timestamps in UTC only in the DB. So please change to Time.now to remain consistent.